### PR TITLE
Replication tunable consistency tests

### PR DESCRIPTION
**Pull request by parkerduckworth**

### CODE REVIEW

File: replication_tunable_consistency.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+set -e
+
+function wait_weaviate() {
+  echo "Wait for Weaviate to be ready"
+  for _ in {1..120}; do
+    if curl -sf -o /dev/null localhost:$1; then
```
Comment (redouan-rhazouani) [inline replication_tunable_consistency.sh:8]: Maybe `if curl -sf -o /dev/null localhost:$1/v1/.well-known/ready` would be more appropriate to check for readiness

File: apps/replication_tunable_consistency/cluster_healthy.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"github.com/weaviate/weaviate/entities/models"
+)
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	log.Printf("Validate all objects were added, with consistency level ONE...")
+
+	b, err := os.ReadFile("data/data.json")
```
Comment (redouan-rhazouani) [inline apps/replication_tunable_consistency/cluster_healthy.go:20]: Maybe we can refactor reading the objects into a function and reuse it the other files

File: replication_tunable_consistency.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      echo "Weaviate is ready"
+      break
+    fi
+
+    echo "Weaviate is not ready on $1, trying again in 1s"
+    sleep 1
+  done
+}
+
+echo "Building all required containers"
```
Comment (redouan-rhazouani) [inline replication_tunable_consistency.sh:18]: I like the testing structure 👍

File: replication_tunable_consistency.sh (hunk 1 above)
Comment (parkerduckworth) [reply replication_tunable_consistency.sh]: Agreed, changed accordingly

File: apps/replication_tunable_consistency/cluster_healthy.go (hunk 2 above)
Comment (parkerduckworth) [reply apps/replication_tunable_consistency/cluster_healthy.go]: Good idea, done

File: replication_tunable_consistency.sh (hunk 3 above)
Comment (parkerduckworth) [reply replication_tunable_consistency.sh:18]: Thanks!

**Review by redouan-rhazouani — APPROVED** [review]

**Pull request merged**